src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/legaladvisor/LegalOrderService.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.hmcts.cmc.claimstore.services.staff.content.legaladvisor;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.ccd.domain.CCDDocument;
+import uk.gov.hmcts.cmc.claimstore.config.properties.pdf.DocumentTemplates;
+import uk.gov.hmcts.cmc.claimstore.documents.PrintService;
+import uk.gov.hmcts.cmc.claimstore.services.document.DocumentManagementService;
+import uk.gov.hmcts.cmc.claimstore.utils.CCDCaseDataToClaim;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimDocumentType;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.sendletter.api.Document;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+@ConditionalOnProperty(prefix = "document_management", name = "url")
+public class LegalOrderService {
+
+    private final DocumentTemplates documentTemplates;
+    private final LegalOrderCoverSheetContentProvider legalOrderCoverSheetContentProvider;
+    private final PrintService legalOrderBulkPrintService;
+    private final DocumentManagementService documentManagementService;
+    private final CCDCaseDataToClaim ccdCaseDataToClaim;
+
+    @Autowired
+    public LegalOrderService(
+        DocumentTemplates documentTemplates,
+        PrintService legalOrderBulkPrintService,
+        LegalOrderCoverSheetContentProvider legalOrderCoverSheetContentProvider,
+        DocumentManagementService documentManagementService,
+        CCDCaseDataToClaim ccdCaseDataToClaim) {
+        this.documentTemplates = documentTemplates;
+        this.legalOrderBulkPrintService = legalOrderBulkPrintService;
+        this.legalOrderCoverSheetContentProvider = legalOrderCoverSheetContentProvider;
+        this.documentManagementService = documentManagementService;
+        this.ccdCaseDataToClaim = ccdCaseDataToClaim;
+    }
+
+    private Document downloadLegalOrder(String authorisation, CCDDocument ccdLegalOrder) throws URISyntaxException {
```

Comment:
We should have first public method followed by private's?

Reply:
sure 👍